### PR TITLE
Added ' to characters allowed in the username

### PR DESCRIPTION
**Pull request by mdusher**
Adds `'` to characters allowed in the username as per #20597

### CODE REVIEW

**Comment by PVince81** [comment]
@blizzz @LukasReschke

**Comment by PVince81** [comment]
To test:
- [ ] TEST: basic auth (curl/cadaver)
- [ ] TEST: mobile clients
- [ ] TEST: desktop clients
- [ ] TEST: web UI login
- [ ] TEST: web UI components where user names are involved
   - [ ] TEST: users page can still list users / set/unset from groups
   - [ ] TEST: user search in share dialog
   - [ ] TEST: user search, applicable field in external storage settings

File: lib/private/user/manager.php [hunk 1]
```diff
@@ -255,9 +255,9 @@ public function createUser($uid, $password) {
 		$l = \OC::$server->getL10N('lib');
 		// Check the name for bad characters
 		// Allowed are: "a-z", "A-Z", "0-9" and "_.@-"
-		if (preg_match('/[^a-zA-Z0-9 _\.@\-]/', $uid)) {
+		if (preg_match('/[^a-zA-Z0-9 _\.@\-\']/', $uid)) {
 			throw new \Exception($l->t('Only the following characters are allowed in a username:'
-				. ' "a-z", "A-Z", "0-9", and "_.@-"'));
+				. ' "a-z", "A-Z", "0-9", and "_.@-"\''));
```
Comment (MorrisJobke) [inline lib/private/user/manager.php]: The `\'` needs to be before the closing `"` ;)

**Comment by MorrisJobke** [comment]
Tested and all works just nice
- [x] TEST: basic auth (curl/cadaver)
- [x] TEST: mobile clients (⚠️I couldn't test the iOS client)
- [x] TEST: desktop clients
- [x] TEST: web UI login
- [x] TEST: web UI components where user names are involved
   - [x] TEST: users page can still list users / set/unset from groups
   - [x] TEST: user search in share dialog
   - [x] TEST: user search, applicable field in external storage settings
- [x] TEST: share via federated cloud sharing to a user with a `'` in the name

:+1: from my side once my comment is addressed

**Comment by MorrisJobke** [comment]
@mdusher Sorry for the delay - this seems to be missed :( Sorry for that.

Once at it - can you please rebase on current master? This makes testing easier ;) Thanks

**Comment by MorrisJobke** [comment]
We currently have issues with test runs of external PRs - I created a local branch (with my comments addressed) to let jenkins run on this code: https://github.com/owncloud/core/pull/21806

**Comment by LukasReschke** [comment]
Let's merge https://github.com/owncloud/core/pull/21806 which has the addressed comment from @MorrisJobke as well.

**Pull request closed without merging**